generators/app/index.js

```diff
@@ -91,19 +124,68 @@ module.exports = yeoman.Base.extend({
         type: 'input',
         name: constants.PROP_PROJECT_GROUP_ID,
         message: 'Project groupId?',
-        default: this.config.get(constants.PROP_PROJECT_GROUP_ID),
+        default: this._getConfigValue(constants.PROP_PROJECT_GROUP_ID),
+        when: function(props) {
+          if(this.bail) return false;
+          return true;
+        }.bind(this),
       },
       {
         type: 'input',
         name: constants.PROP_PROJECT_ARTIFACT_ID,
+        when: function(props) {
+          if(this.bail) return false;
+          return true;
+        }.bind(this),
+        default: this._getConfigValue(constants.PROP_PROJECT_ARTIFACT_ID),
         message: 'Project artifactId?',
-        default: this.config.get(constants.PROP_PROJECT_ARTIFACT_ID),
+      },
+      {
+        // When run inside a existing project
+        type: 'confirm',
+        name: constants.PROP_ABORT_PROJECT_ARTIFACT_ID_UPDATE,
+        when: function(props) {
+          if(this.bail) return false;
+          if(fs.existsSync(this.config.path) && props[constants.PROP_PROJECT_ARTIFACT_ID] != this._getConfigValue(constants.PROP_PROJECT_ARTIFACT_ID)) {
+            return true;
+          }
+          return false;
+        }.bind(this),
+        default: true,
+        message: 'Updating the artifactId can lead to unexpected issues. Do you want to abort?',
+      },
+      {
+        // When run inside a non-existing project
+        type: 'confirm',
+        name: constants.PROP_CREATE_SUB_FOLDER,
+        when: function(props) {
+          if(props[constants.PROP_ABORT_PROJECT_ARTIFACT_ID_UPDATE]) {
+            this.bail = true;
+          }
+          if(this.bail) return false;
+          if(!fs.existsSync(this.config.path) && props[constants.PROP_PROJECT_ARTIFACT_ID] != this._getConfigValue(constants.PROP_PROJECT_ARTIFACT_ID)) {
+            props[constants.PROP_PROJECT_PATH] = path.join(process.cwd(),props[constants.PROP_PROJECT_ARTIFACT_ID]);
+            this.out.warn('The artifactId should be identical to the folder name. We are going to generate the project inside the folder "'
+                 +path.join(process.cwd(),props[constants.PROP_PROJECT_ARTIFACT_ID])+'".');
+            return true;
+          }
+          return false;
+        }.bind(this),
+        default: true,
+        message: 'Do you want to proceed?',
       },
       {
         type: 'input',
         name: constants.PROP_PROJECT_VERSION,
+        when: function(props) {
+          if(props[constants.PROP_CREATE_SUB_FOLDER] != null && !props[constants.PROP_CREATE_SUB_FOLDER]) {
```

Comment:
In "props[constants.PROP_CREATE_SUB_FOLDER] != null", I was trying to handle both null and undefined at the same time. I agree it is better to have CONST <operator> VALUE. Do you think it is better to write "null !== props[constants.PROP_CREATE_SUB_FOLDER] && undefined !== props[constants.PROP_CREATE_SUB_FOLDER]" instead of "null != props[constants.PROP_CREATE_SUB_FOLDER]"